File: Python-For-Everyone-Horstmann/Chapter5-Functions/P5.17.py
# Use recursion to implement a function  find(string, match) that tests whether  match is
# contained in  string :
#     b = find("Mississippi", "sip") #  Sets  b to  true
# Hint: If  string starts with  match , you are done. If not, consider the string that you
# obtain by removing the first character.

# FUNCTIONS
def find(string, match):
    if len(string) < len(match):
        return False

    if string[:len(string)] == match:
        return True

    end = int(1 + len(string) - 1)

    return find(string[1:end], match)

def main():
    inputString = str(input("Enter the string: "))
    inputMatch = str(input("Enter the string to find: "))

    print("Found?", find(inputString, inputMatch))

# PROGRAM RUN
main()

File: Big-Java-for-Everyone-Late-Objects-Horstmann/Chapter-5-Methods/P5_17.java
/*Use recursion to implement a method public static boolean find(String str, String
match) that tests whether match is contained in str :
boolean b = find("Mississippi", "sip"); // Sets b to true
Hint: If str starts with match , then you are done. If not, consider the string that you
obtain by removing the first character.*/

public class P5_17 {
    public static boolean find(String str, String match) {
        if (str.length() < match.length()) {
            return false;
        }
        if (str.charAt(0) == match.charAt(0)) {
            boolean equal = true;
            for (int i = 0; i < match.length(); i++) {
                if (str.charAt(i) != match.charAt(i)) {
                    equal = false;
                }
            }
            if (equal) {
                return true;
            }
        }
        return find(str.substring(1), match);
    }
}
